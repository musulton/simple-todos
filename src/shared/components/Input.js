import React from 'react';
import {TextInput, View, StyleSheet} from "react-native";

const Input = ({inputValue, onInputChange, placeholder}) => (
    <View style={styles.inputContainer}>
        <TextInput
            value={inputValue}
            style={styles.input}
            placeholder={placeholder}
            placeholderTextColor='#CACACA'
            selectionColor='#666666'
            onChangeText={onInputChange}
        />
    </View>)

const styles = StyleSheet.create({
    inputContainer: {
        marginLeft: 20,
        marginRight: 20,
        //Shadowing for IOS
        shadowOpacity: 0.2,
        shadowRadius: 3,
        shadowColor: '#000000',
        shadowOffset: {width: 3, height: 3},
        //Shadowing for Android
        elevation: 3,
        //Mandatory style for shadowing
        backgroundColor: '#ffffff',
    }, input: {
        height: 60,
        backgroundColor: '#ffffff',
        paddingLeft: 10,
        paddingRight: 10
    }
});
export default Input;
