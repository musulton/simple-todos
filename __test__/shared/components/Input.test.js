import {fireEvent, render, screen} from "@testing-library/react-native";

import Input from "../../../src/shared/components/Input";

describe("Input", () => {
    test("should render correctly", () => {
        const mockProps = {
            inputValue: "Value",
            onInputChange: jest.fn(),
            placeholder: "Placeholder"
        }

        render(<Input {...mockProps} />)

        expect(screen.toJSON()).toMatchSnapshot()
    })

    test("should call onInputChange when change input text", () => {
        const mockProps = {
            inputValue: "Value",
            onInputChange: jest.fn(),
            placeholder: "Placeholder"
        }
        const inputValue = "input value"

        render(<Input {...mockProps} />)
        fireEvent.changeText(screen.getByPlaceholderText(mockProps.placeholder), inputValue)

        expect(mockProps.onInputChange).toHaveBeenCalledWith(inputValue)
    })
})
