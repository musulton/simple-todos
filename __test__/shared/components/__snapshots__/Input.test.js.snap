// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#ffffff",
      "elevation": 3,
      "marginLeft": 20,
      "marginRight": 20,
      "shadowColor": "#000000",
      "shadowOffset": {
        "height": 3,
        "width": 3,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 3,
    }
  }
>
  <TextInput
    onChangeText={[MockFunction]}
    placeholder="Placeholder"
    placeholderTextColor="#CACACA"
    selectionColor="#666666"
    style={
      {
        "backgroundColor": "#ffffff",
        "height": 60,
        "paddingLeft": 10,
        "paddingRight": 10,
      }
    }
    value="Value"
  />
</View>
`;
